src/datastore/access-policies_test.cpp

```diff
@@ -177,3 +177,18 @@ TEST_F(AccessPoliciesTest, store) {
 		R"([{"attrs": "{\"key\":\"value\"}", "resource": "resource/AccessPoliciesTest.store"}])",
 		rules);
 }
+
+TEST_F(AccessPoliciesTest, retrieveByCollection) {
+	const datastore::Collection   collection({
+		  .name = "collection_name:AccessPoliciesTest.retrieveByCollection",
+    });
+	const datastore::AccessPolicy access({
+		.name = "policy_id:AccessPoliciesTest.retrieveByCollection",
+	});
+
+	EXPECT_NO_THROW(collection.store());
+	EXPECT_NO_THROW(access.store());
+	EXPECT_NO_THROW(access.addCollection(collection.id()));
```

Comment:
It's better to use `ASSERT_*` instead of `EXPECT_*` for setup steps where you wouldn't want the test to continue executing after a failure.